addons/osfstorage/views.py

```diff
@@ -85,7 +85,12 @@ def osfstorage_get_revisions(file_node, node_addon, payload, **kwargs):
 
 @decorators.waterbutler_opt_hook
 def osfstorage_copy_hook(source, destination, name=None, **kwargs):
-    return source.copy_under(destination, name=name).serialize(), httplib.CREATED
+    try:
+        return source.copy_under(destination, name=name).serialize(), httplib.CREATED
+    except exceptions.FileNodeIsQuickFilesNode:
+        raise HTTPError(httplib.METHOD_NOT_ALLOWED, data={
```

Comment:
"405 Method not allowed" is returned when the **HTTP method** is not valid. In this case, the method is supported ("POST"), but the request is semantically invalid, i.e. our business rules prevent copying to quickfiles. Therefore, a 400 is more appropriate here.